src/components/visualization/layout/ChartTableWrapper.tsx

```diff
@@ -22,16 +25,40 @@ export default function ChartTableWrapper({
   transformedData: DataRecord;
 }) {
   const t = useTranslations('ChartTableWrapper');
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const router = useRouter();
   const [filteredData, setFilteredData] = useState(transformedData);
   const { width, height } = useWindowDimensions();
-  const resourceArr = Object.entries(resource.visualizations);
+
+  const setVisualizationParameter = useDebouncedCallback((eventKey: string) => {
+    const params = new URLSearchParams(searchParams);
+    params.set('visualization', eventKey);
+    router.replace(`${pathname}?${params.toString()}`);
+  });
+  const [activeVisualization, setActiveVisualization] = useState(getDefaultVisualization());
+  function getDefaultVisualization() {
```

Comment:
Rename this to something like `getAndUpdateVisualization`, or `getAndUpdateVisualizationParameter`, to make it clear that this function is not a read-only function

Reply:
Good point. Fixed.